src/VolumeLogger.js

```diff
@@ -59,6 +59,295 @@ module.exports = class VolumeLogger {
             name: 'messageQueueSize'
         })
 
+        this.stopped = false
+
+        if (this.client instanceof StreamrClient) {
+            let sec = 0
+
+            const throtheledAvg = (avg, avgInterval) => {
+                return 0.9 * avg + 0.2 * avgInterval
+            }
+
+            const securePublish = async (stream, data) => {
+                if (this.stopped) {
+                    return
+                }
+                this.client.publish(stream, data)
+            }
+
+            const getResend = async (stream, last, timeout = 1000) => {
+                return new Promise((resolve, reject) => {
+                    const startTimeout = () => {
+                        return setTimeout(() => {
+                            resolve([])
+                        }, timeout)
+                    }
+
+                    let timeoutId = startTimeout()
+                    const messages = []
+                    if (this.stopped) {
+                        resolve(messages)
+                    } else {
+                        this.client.resend(
+                            {
+                                stream,
+                                resend: {
+                                    last
+                                }
+                            },
+                            (message) => {
+                                messages.push(message)
+                                clearTimeout(timeoutId)
+                                timeoutId = startTimeout()
+                                if (messages.length === last) {
+                                    resolve(messages)
+                                }
+                            }
+                        )
+                    }
+                })
+            }
+
+            const minReport = {
```

Comment:
Is there a particular reason `minReport` is initialized differently than the others?